### PR TITLE
OOP school library: unit tests

### PR DESCRIPTION
**Pull request by musangamfure**
**Hello Reviewer👋,** 

In this PR, we have worked on the Unit Testing feature using the RSpec framework.

### **Here is the test we have done:**

- `Book` class
- `Person` class
- `Student` class
- `Teacher` class
- `Classroom` class
- `Rental` class

Kindly check our codes and let us know what you think.

**Happy Coding🎉🎉**

### CODE REVIEW

File: book_spec.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      rental = book.add_rental(date, person)
+
+      expect(rental).to be_instance_of(Rental)
+      expect(rental.book).to eq(book)
+      expect(rental.person).to eq(person)
+      expect(book.rentals).to include(rental)
+      expect(person.rentals).to include(rental)
+    end
+  end
+end
```
Comment (tresorsawasawa) [inline book_spec.rb]: - [ ] As required for the project, kindly ensure that all test files are placed in a different directory (`e.g spec/`). I suggest you create a directory named `spec/` and move all your test files insitde.

File: person_spec.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+  describe '#add_rental' do
+    let(:rental) { instance_double('Rental') }
+
+    it 'adds a rental to the person and sets the person on the rental' do
+      expect(rental).to receive(:person=).with(person)
+      person.add_rental(rental)
+      expect(person.rentals).to include(rental)
+    end
+  end
```
Comment (tresorsawasawa) [inline person_spec.rb]: - [ ] As required for the project to add unit tests for all the methods in all the classes, kindly ensure that you also add tests to the `of_age?` method as the only missing in the `Person` calss.

File: book_spec.rb (hunk 1 above)
Comment (tresorsawasawa) [inline book_spec.rb]: - [ ] Also as it is required for the project  to add tests to all classes that you created while implementing the Decorator design pattern(e.g. : `BaseDecorator`, `CapitalizeDecorator`, `TrimmerDecorator`, `Nameable` ), kindly ensure that you also create specific file to each one and add test to each method inside

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @musangamfure and @lordksix 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow Gitflow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book_spec.rb (hunk 1 above)
Comment (rloterh) [reply book_spec.rb]: well resolved! ✔️

File: person_spec.rb (hunk 2 above)
Comment (rloterh) [reply person_spec.rb]: well resolved! ✔️

File: book_spec.rb (hunk 1 above)
Comment (rloterh) [reply book_spec.rb]: well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Emmanuel & Wladimir,  

Your project is complete! There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 🎯 
🎲 Good job for introducing improvements to the specs!
🎲 Professional ReadMe!

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**